server/src/main/java/org/apache/iotdb/db/mpp/execution/QueryExecution.java

```diff
@@ -300,9 +300,7 @@ private void initResultHandle() {
               .createSourceHandle(
                   context.getResultNodeContext().getVirtualFragmentInstanceId().toThrift(),
                   context.getResultNodeContext().getVirtualResultNodeId().getId(),
-                  new TEndPoint(
-                      context.getResultNodeContext().getUpStreamEndpoint().getIp(),
-                      IoTDBDescriptor.getInstance().getConfig().getDataBlockManagerPort()),
+                  context.getLocalDataBlockEndpoint(),
```

Comment:
The result handle in coordinator is sourceHanlde, endPoint of it should be upStreamEndPoint, not its own endPoint.
It shoud be `context.getResultNodeContext().getUpStreamEndpoint()`

Reply:
Fixed